src/sentry/integrations/github/client.py

```diff
@@ -1,61 +1,109 @@
 from __future__ import annotations
 
 from datetime import datetime
+from typing import Any, Mapping, Sequence
 
 import sentry_sdk
+from rest_framework.response import Response
 
 from sentry.integrations.client import ApiClient
 from sentry.integrations.github.utils import get_jwt
-from sentry.models import Repository
+from sentry.models import Integration, Repository
 from sentry.utils import jwt
+from sentry.utils.json import JSONData
 
 
-class GitHubClientMixin(ApiClient):
-    allow_redirects = True
+def get_next_link(resp: Response) -> str | None:
+    link_option: str | None = resp.headers.get("link")
+    if link_option is None:
+        return None
 
-    base_url = "https://api.github.com"
-    integration_name = "github"
+    # Should be a comma separated string of links
+    links = link_option.split(",")
 
-    def get_jwt(self):
-        return get_jwt()
+    for link in links:
+        # If there is a 'next' link return the URL between the angle brackets, or None
+        if 'rel="next"' in link:
+            start = link.find("<") + 1
+            end = link.find(">")
+            return link[start:end]
 
-    def get_last_commits(self, repo, end_sha):
-        # return api request that fetches last ~30 commits
-        # see https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
-        # using end_sha as parameter
-        return self.get_cached(f"/repos/{repo}/commits", params={"sha": end_sha})
+    return None
 
-    def compare_commits(self, repo, start_sha, end_sha):
-        # see https://developer.github.com/v3/repos/commits/#compare-two-commits
-        # where start sha is oldest and end is most recent
-        return self.get_cached(f"/repos/{repo}/compare/{start_sha}...{end_sha}")
 
-    def repo_hooks(self, repo):
-        return self.get(f"/repos/{repo}/hooks")
+class GitHubClientMixin(ApiClient):  # type: ignore
+    allow_redirects = True
 
-    def get_commits(self, repo):
-        return self.get(f"/repos/{repo}/commits")
+    base_url = "https://api.github.com"
+    integration_name = "github"
 
-    def get_commit(self, repo, sha):
-        return self.get_cached(f"/repos/{repo}/commits/{sha}")
+    def get_jwt(self) -> str:
+        return get_jwt()
 
-    def get_repo(self, repo):
-        return self.get(f"/repos/{repo}")
+    def get_last_commits(self, repo: str, end_sha: str) -> Sequence[JSONData]:
+        """
+        Return API request that fetches last ~30 commits
+        see https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
+        using end_sha as parameter.
+        """
+        # Explicitly typing to satisfy mypy.
```

Comment:
I don't know if you need this comment in every function.

Reply:
You're right, I'll stop leaving the comment when the file is in `mypy.ini`.